pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -765,6 +744,8 @@ func (r *MachineSetSyncReconciler) createOrUpdateCAPIInfraMachineTemplate(ctx co
 }
 
 // createOrUpdateCAPIMachineSet creates a CAPI machine set from a MAPI one, or updates if it exists and it is out of date.
+//
+//nolint:funlen,gocognit,cyclop
```

Comment:
Are we at the point where it makes sense to split this function into its composite parts? 😓 


One for creation? One for updating? One (or more?) for handling status?